### PR TITLE
activerecord-hackery/ransack#374 Searching across multiple associatio…

### PR DESCRIPTION
**Pull request by bertBruynooghe**
…ns that reference the same model/table joining on the wrong association

### CODE REVIEW

**Comment by bertBruynooghe** [comment]
'Solved' the issue by dynamically creating associations on the models so the association code is solving the queries. Although it solves the issues I had and the tests that were introduced, I don't think it covers all code paths (polymorphic and or expressions are currently excluded), and it might render a few code paths into dead code...

**Comment by jonatack** [comment]
@bertBruynooghe Thanks! Unfortunately travis-ci is red everywhere. Did the specs pass for you in your development environment?

**Comment by bertBruynooghe** [comment]
The rails specs and ransack tests passed individually…

Regards,
Bert Bruynooghe.

**Comment by jonatack** [comment]
OK, will give it a try in my local env.

File: spec/ransack/search_spec.rb [hunk 1]
```diff
@@ -241,17 +240,23 @@ module Ransack
...
+           WHERE \\(+people.name = 'Ernie' AND (\\S+).name = 'Test'\\)+
+          SQL
+
+        match = Regexp.new(expected_query_exp.squish).match(real_query)
+
+        expect(match[1]).to eq match[2]
+        expect(match[1]).to eq match[5]
+        expect(match[3]).to eq match[4]
+        expect(match[3]).to eq match[6]
+        expect(match[1]).not_to eq match[3]
```
Comment (avit) [inline spec/ransack/search_spec.rb]: We could do these assertions with Regexp backreferences.

Can we get a test here that executes the query and validates results to see it's valid?

**Comment by avit** [comment]
There are only a few failing tests (but for every version it seems). It looks like something unrelated was broken as part of this change.

**Comment by seanfcarroll** [comment]
I know this is old. If we want to merge it, it will need test coverage for the new methods.

**Pull request closed without merging**